web_api.md

```diff
@@ -0,0 +1,337 @@
+for Python. [RabbitMQ](https://www.rabbitmq.com/) is used for the Process Queue, and [MongoDB](https://www.mongodb.com/)
+is the database system. There are many options for a reverse proxy, such as Nginx, Apache, or HAProxy. From our side, we
+recommend using [Traefik](https://doc.traefik.io/traefik/).
+
+### Description
+
+As shown in Fig. 1, each section in the [Web API specification](#the-specification) is implemented by different servers,
+which are Discovery Server, Processing Server, Workflow Server, and Workspace Server respectively. Although each server
+in the figure is deployed on its own machine, it is completely up to the implementors to decide which machines run which
+servers. However, having each processor run on its own machine reduces the risk of version and resource conflicts.
+Furthermore, the machine can be customized to best fit the processor's hardware requirements and throughput demand. For
+example, some processors need GPU computation, while others do not, or some need more CPU capacity while others need
+more memory.
+
+**Processing**: since the `Processing` section is provided by [OCR-D Core](https://github.com/OCR-D/core), implementors
+do not need to implement Processing Server, Process Queue, and Processing Worker themselves, they can reuse/customize
+the existing implementation. Once a request arrives, it will be (split into single-processor jobs and) pushed to a
+job queue. A job queue always has the same name as its respective processors. For example, `ocrd-olena-binarize`
+processors listen only to the queue named `ocrd-olena-binarize`. A Processing Worker, which is
+an [OCR-D Processor](https://ocr-d.de/en/spec/glossary#ocr-d-processor) running as a worker, listens to the queue, pulls
+new jobs when available, processes them, and push the job statuses back to the queue if necessary. One normally does not
+run a Processing Worker directly, but via a Processing Server. Job statuses can be pushed back to the queue, depending
+on the [job configuration](#process-queue), so that other services get updates and act accordingly.
+
+**Database**: in this architecture, a database is required to store information such as users requests, jobs statuses,
+workspaces, etc. [MongoDB](https://www.mongodb.com/) is required here.
+
+**Network File System**: in order to avoid file transfer between different machines, it is highly recommended to have
+a [Network File System (NFS)](https://en.wikipedia.org/wiki/Network_File_System) set up. With NFS, all Processing
+Servers(specifically processors) can work in a shared storage environment and access files as if they are local files.
+To get data into the NFS, one could use the `POST /workspace` endpoint to
+upload [OCRD-ZIP](https://ocr-d.de/en/spec/ocrd_zip)files. However, this approach is only appropriate for testing or
+very limited data sizes. Usually, Workspace Server should be able to pull data from other storages.
+
```

Comment:
Perhaps add a 3rd level section on the Workflow Server (along the lines of the new entry in the terminology) here?

Reply:
I will do it later, when we have the working example Nextflow script.